src/traces/pie/helpers.js

```diff
@@ -20,7 +20,9 @@ exports.formatPiePercent = function formatPiePercent(v, separators) {
 
 exports.formatPieValue = function formatPieValue(v, separators) {
     var vRounded = v.toPrecision(10);
-    if(vRounded.lastIndexOf('.') !== -1) {
+    if(vRounded.indexOf('e+') !== -1) {
+        vRounded = Math.round(v);
+    } else if(vRounded.lastIndexOf('.') !== -1) {
```

Comment:
There's also `e-` if the number is <1e-6

Can we do something like:
```
if(vRounded.indexOf('e') !== -1) {
    vRounded = vRounded.replace(/[.]?0+e/, 'e');
} else if(vRounded.indexOf('.') !== -1) { // why did we have lastIndexOf here? indexOf should be the same, right?
```

Reply:
Good call. Revised in https://github.com/plotly/plotly.js/pull/5152/commits/58916511f71c2f6c58364735f5acc95b49687246.